### PR TITLE
IPv6 for microserver and microDNS

### PR DESCRIPTION
**Pull request by macisasandwich**

### CODE REVIEW

**Comment by scw00** [comment]
[approve ci]

File: tests/gold_tests/autest-site/microDNS.test.ext [hunk 1]
```diff
@@ -86,21 +86,32 @@ def MakeDNServer(obj, name, filename="dns_file.json", port=False, IP='127.0.0.1'
         with open(filepath, 'w') as f:
             f.write(json.dumps(jsondata))
 
+    if ipv6:
```
Comment (persiaAziz) [inline tests/gold_tests/autest-site/microDNS.test.ext]: umm can we pass IP as an option as well, rather than hardcoding it

**Comment by dragon512** [comment]
[approve ci]

File: tests/gold_tests/autest-site/microDNS.test.ext [hunk 2]
```diff
@@ -68,7 +68,7 @@ def addRecords(self, records=None, jsonFile=None):
         f.write(json.dumps(jsondata))
 
 
-def MakeDNServer(obj, name, filename="dns_file.json", port=False, IP='127.0.0.1', rr=False, default=None, options={}):
+def MakeDNServer(obj, name, filename="dns_file.json", port=False, IP=False, rr=False, default=None, ipv6=False, options={}):
```
Comment (persiaAziz) [inline tests/gold_tests/autest-site/microDNS.test.ext]: why is IP a boolean here and a IP string in line 89? :) 

my suggestion was 
```
def MakeDNServer(obj, name, filename="dns_file.json", port=False, IP='::1',....
...
.... 
IP = IP
```

Comment (persiaAziz) [reply tests/gold_tests/autest-site/microDNS.test.ext]: the idea to allow microDNS run on a different host if necessary

**Comment by persiaAziz** [comment]
[approve CI]

File: tests/gold_tests/autest-site/microDNS.test.ext [hunk 3]
```diff
@@ -68,7 +68,7 @@ def addRecords(self, records=None, jsonFile=None):
         f.write(json.dumps(jsondata))
 
 
-def MakeDNServer(obj, name, filename="dns_file.json", port=False, IP='127.0.0.1', rr=False, default=None, options={}):
+def MakeDNServer(obj, name, filename="dns_file.json", port=False, ip='127.0.0.1', rr=False, default=None, ipv6=False, options={}):
```
Comment (SolidWallOfCode) [inline tests/gold_tests/autest-site/microDNS.test.ext]: This seems a bad API design choice to me. The family should be taken from the `ip` parameter, not specified separately. If you want to allow selection of special addresses by family, use the same technique as in C/C++ - reserve keywords such as "IP_LOOPBACK" or "IP_ADDR_ANY" to indicate such things.

File: tests/tools/lib/IPConstants.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import ipaddress
+
+# convenience functions
+
+IPkw = {'INADDR_LOOPBACK':'127.0.0.1',
```
Comment (persiaAziz) [inline tests/tools/lib/IPConstants.py]: is having keywords like this absolutely necessary?

**Pull request merged**